tests/unit/replication/ReplicationStatusProcessor.js

```diff
@@ -0,0 +1,138 @@
+const promClient = require('prom-client');
+
+// Clear register to avoid:
+// Error: A metric with the name kafka_lag has already been registered.
+promClient.register.clear();
+
+const ReplicationStatusProcessor =
+    require('../../../extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor');
+
+function makeReplicationStatusProcessor(replayTopics) {
+    return new ReplicationStatusProcessor(
+        { hosts: 'localhost:9092' },
+        {
+            auth: {},
+            s3: {},
+            transport: 'http',
+        },
+        { replayTopics },
+        {});
+}
+
+describe('ReplicationStatusProcessor', () => {
+    describe('::_reshapeReplayTopics', () => {
+        it('should return undefined if the config replay topics is undefined', () => {
+            const replayTopics = undefined;
+            const replicationStatusProcessor = makeReplicationStatusProcessor(replayTopics);
+            const stateVars = replicationStatusProcessor.getStateVars();
+
+            assert.strictEqual(stateVars.replayTopics, undefined);
+        });
+
+        it('should return undefined if the config replay topics is empty array', () => {
+            const replayTopics = [];
+            const replicationStatusProcessor = makeReplicationStatusProcessor(replayTopics);
+            const stateVars = replicationStatusProcessor.getStateVars();
+
+            assert.strictEqual(stateVars.replayTopics, undefined);
+        });
+
+        it('should return an array of five items if the config replay topic has five retries', () => {
+            const replayTopics = [
+                {
+                    topicName: 'backbeat-replication-replay-0',
+                    retries: 5,
+                },
+            ];
+            const replicationStatusProcessor = makeReplicationStatusProcessor(replayTopics);
+            const stateVars = replicationStatusProcessor.getStateVars();
+
+            assert.deepStrictEqual(stateVars.replayTopics, [
+                'backbeat-replication-replay-0',
+                'backbeat-replication-replay-0',
+                'backbeat-replication-replay-0',
+                'backbeat-replication-replay-0',
+                'backbeat-replication-replay-0',
+            ]);
+        });
+
+        it('should return an array of seven items if the config replay topic has seven retries', () => {
+            const replayTopics = [
+                {
+                    topicName: 'backbeat-replication-replay-0',
+                    retries: 5,
+                },
+                {
+                    topicName: 'backbeat-replication-replay-1',
+                    retries: 2,
+                },
+            ];
+            const replicationStatusProcessor = makeReplicationStatusProcessor(replayTopics);
+            const stateVars = replicationStatusProcessor.getStateVars();
+
+            assert.deepStrictEqual(stateVars.replayTopics, [
+                'backbeat-replication-replay-1',
```

Comment:
I see why it's in reverse, but a comment either here or in the function definitions explaining why would save people time :)